### ISSUE TITLE
or-fun-call macro suggestion bug

### ISSUE THREAD

**Issue by lrbalt** [rust-lang/rust-clippy#1018]
I have this code:

``` rust
r.view_args
     .get(name)
     .ok_or(format!("{} not given", name))
```

Clippy reports: 

```
src/app/app_server.rs:259:5: 261:43 warning: use of `ok_or` followed by a function call, #[warn(or_fun_call)] on by default
src/app/app_server.rs:259     r.view_args
                              ^
src/app/app_server.rs:259:5: 261:43 help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#or_fun_call
src/app/app_server.rs:259:5: 261:43 help: try this
src/app/app_server.rs:        r.view_args
src/app/app_server.rs:         .get(name).ok_or_else(|| $ crate :: fmt :: format ( format_args ! ( $ ( $ arg ) * ) ))
```

Perhaps Clippy should ignore `.ok_or(macro!())` constructs?

**Comment by mcarton** [comment]
The suggestion is badly formatted but `ok_or_else` would still be a better choice here.

**Comment by llogiq** [comment]
Is this a problem with `snippet(..)`?

**Comment by mcarton** [comment]
Yes, it's been there since #944 with some macro spans.

**Comment by lrbalt** [comment]
Do you mean that
```rust
.view_args
     .get(name)
     .ok_or_else(format!("{} not given", name))
```
is better here?

**Comment by mcarton** [comment]
```rust
     .ok_or_else(|| format!("{} not given", name))
```
It avoids build the string if it is not necessary.

**Comment by lrbalt** [comment]
ok, thanks
